launcher_go/v2/docker/commands.go

```diff
@@ -0,0 +1,54 @@
+package docker
+
+import (
+	"context"
+	"github.com/discourse/discourse_docker/launcher_go/v2/config"
+	"github.com/discourse/discourse_docker/launcher_go/v2/utils"
+	"golang.org/x/sys/unix"
+	"io"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+type DockerBuilder struct {
+	Config   *config.Config
+	Ctx      *context.Context
+	Stdin    io.Reader
+	Dir      string
+	ImageTag string
+}
+
+func (r *DockerBuilder) Run() error {
+	if r.ImageTag == "" {
+		r.ImageTag = "latest"
+	}
+	cmd := exec.CommandContext(*r.Ctx, utils.DockerPath, "build")
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	cmd.Cancel = func() error {
+		return unix.Kill(-cmd.Process.Pid, unix.SIGINT)
+	}
+	cmd.Dir = r.Dir
+	cmd.Env = r.Config.EnvArray(false)
+	cmd.Env = append(cmd.Env, "BUILDKIT_PROGRESS=plain")
+	for k, _ := range r.Config.Env {
+		cmd.Args = append(cmd.Args, "--build-arg")
+		cmd.Args = append(cmd.Args, k)
+	}
+	cmd.Args = append(cmd.Args, "--no-cache")
+	cmd.Args = append(cmd.Args, "--pull")
+	cmd.Args = append(cmd.Args, "--force-rm")
+	cmd.Args = append(cmd.Args, "-t")
+	cmd.Args = append(cmd.Args, utils.BaseImageName+r.Config.Name+":"+r.ImageTag)
+	cmd.Args = append(cmd.Args, "--shm-size=512m")
```

Comment:
curious why we need to increase the shm size here.

Reply:
the shm size is in current launcher and I carried that option over 1-1 here